File: src/main/java/com/dongfg/project/api/DongfgApiApplication.java
package com.dongfg.project.api;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cache.annotation.EnableCaching;
import org.springframework.data.mongodb.repository.config.EnableMongoRepositories;

/**
 * @author dongfg
 * @date 2017/10/15
 */
@SpringBootApplication
@EnableMongoRepositories
@EnableCaching
public class DongfgApiApplication {

    public static void main(String[] args) {
        SpringApplication.run(DongfgApiApplication.class, args);
    }
}

File: src/main/kotlin/com/dongfg/project/api/DongfgApiApplication.kt
package com.dongfg.project.api

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.FilterType
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.config.EnableJpaRepositories
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.mongodb.repository.config.EnableMongoRepositories

@EnableJpaRepositories(basePackages = ["com.dongfg.project.api.repository"],
        includeFilters = [ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE, value = [(JpaRepository::class)])])
@EnableMongoRepositories(basePackages = ["com.dongfg.project.api.repository"],
        includeFilters = [ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE, value = [(MongoRepository::class)])])
@SpringBootApplication
class DongfgApiApplication

fun main(args: Array<String>) {
    runApplication<DongfgApiApplication>(*args)
}
